app/javascript/gobierto_dashboards/webapp/containers/contracts/Summary.vue

```diff
@@ -176,12 +228,173 @@ export default {
       labelProcessType: I18n.t('gobierto_dashboards.dashboards.contracts.process_type'),
       labelAmountDistribution: I18n.t('gobierto_dashboards.dashboards.contracts.amount_distribution'),
       labelMainAssignees: I18n.t('gobierto_dashboards.dashboards.contracts.main_assignees'),
+      labelBeesWarm: I18n.t('gobierto_dashboards.dashboards.visualizations.title_beeswarm'),
+      labelTooltipBeesWarm: I18n.t('gobierto_dashboards.dashboards.visualizations.tooltip_beeswarm'),
+      labelMultipleLine: I18n.t('gobierto_dashboards.dashboards.visualizations.title_multiple'),
+      queryLineChart: "?sql=SELECT final_amount_no_taxes, status, initial_amount_no_taxes, start_date FROM contratos WHERE contract_type != 'Patrimonial'",
+      dataBeesWarm: undefined,
+      dataBeesWarmFilter: undefined,
+      dataLineChart: undefined,
+      valuesForLineChart: undefined,
+      valuesForCircleChart: undefined,
+      labelRootKey: I18n.t('gobierto_dashboards.dashboards.contracts.assignees'),
     }
   },
-  created() {
+  watch: {
+    dashboardsData(newValue, oldValue) {
+      if (newValue !== oldValue) {
+        this.updateDataBeesWarm(newValue)
+      }
+    }
+  },
+  async created() {
     this.columns = assigneesColumns;
+    this.dataBeesWarmFilter = JSON.parse(JSON.stringify(this.dashboardsData));
+
+    const { data: { data: dataLineChart } } = await getQueryData(this.queryLineChart)
+    this.transformDataContractsLine(dataLineChart)
   },
   methods: {
+    updateDataBeesWarm(data){
+      const dataBeesWarm = JSON.parse(JSON.stringify(data));
+      this.dataBeesWarmFilter = dataBeesWarm
+    },
+    transformDataContractsLine(data) {
+      data.forEach(d => {
+        d.final_amount_no_taxes = +d.final_amount_no_taxes
+        d.initial_amount_no_taxes = +d.initial_amount_no_taxes
+        //Calculate percentage median between initial amount and final amount to obtain the difference.
+        d.year = new Date(d.start_date).getFullYear()
+
+        if (d.status === "Formalizado" || d.status === "Adjudicado") {
+          d.formalized = 1
+        } else {
+          d.formalized = 0
+        }
+
+        if (d.status === "Desierto") {
+          d.anulled = 1
+        } else {
+          d.anulled = 0
+        }
+
+        if (d.final_amount_no_taxes === 0) {
+          d.formalized = 0
+          d.anulled = 1
+        }
+      })
+
+      //JS convert null years to 1970
+      data = data.filter(({ year }) => year !== 1970 && year !== 2021)
+
+      let dataFormalizeContracts = data.filter(({ formalized }) => formalized === 1)
+      dataFormalizeContracts.forEach(d => {
+        d.percentage_total = Math.abs(((d.final_amount_no_taxes - d.initial_amount_no_taxes) / d.initial_amount_no_taxes) * 100)
```

Comment:
Possible division by zero if initial_amount_no_taxes = 0

Reply:
It is not possible because each contract `formalized` has to have a `final_amount_no_taxes` higher than zero.

```
if (d.final_amount_no_taxes === 0) {
  d.formalized = 0
  d.anulled = 1
}
```

Reply:
Still don't understand. I'm talking about `initial_amount_no_taxes`. It's not being modified anywhere, not filtered... it's only parsed as Number. So, it could come as `0`